packages/wattles/lib/src/repository.dart

```diff
@@ -0,0 +1,114 @@
+import 'package:wattles/wattles.dart';
+
+/// {@template repository}
+/// A repository for performing actions on the database of a given [Struct]'s
+/// [Schema].
+/// {@endtemplate}
+abstract class Repository<T extends Struct> {
+  /// {@macro repository}
+  Repository({
+    required String table,
+    required Schema schema,
+    DatabaseDriver? driver,
+  })  : assert(schema is T, 'schema must be of type $T'),
+        _table = table,
+        _rootSchema = schema,
+        _driver = driver ?? MemoryDriver(); // TODO(wolfen): create defaults??
```

Comment:
Since this is abstract, I don't think you need defaults here. I bet people will hardly want to use the memory driver, so better make it required so people don't forget to settle it. wdyt?

Reply:
The defaults TODO is meant to indicate that I have to create a way for people to hook everything up, this well be reworked in my next PR :D